src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/model/ManagedIndexMetaData.kt

```diff
@@ -229,12 +254,23 @@ data class ManagedIndexMetaData(
                     POLICY_PRIMARY_TERM -> policyPrimaryTerm = if (xcp.currentToken() == Token.VALUE_NULL) null else xcp.longValue()
                     POLICY_COMPLETED -> policyCompleted = if (xcp.currentToken() == Token.VALUE_NULL) null else xcp.booleanValue()
                     ROLLED_OVER -> rolledOver = if (xcp.currentToken() == Token.VALUE_NULL) null else xcp.booleanValue()
-                    TRANSITION_TO -> transitionTo = xcp.text()
-                    StateMetaData.STATE -> state = StateMetaData.parse(xcp)
-                    ActionMetaData.ACTION -> action = ActionMetaData.parse(xcp)
-                    StepMetaData.STEP -> step = StepMetaData.parse(xcp)
-                    PolicyRetryInfoMetaData.RETRY_INFO -> retryInfo = PolicyRetryInfoMetaData.parse(xcp)
+                    TRANSITION_TO -> transitionTo = if (xcp.currentToken() == Token.VALUE_NULL) null else xcp.text()
+                    StateMetaData.STATE -> {
+                        // check null for invalid policy situation
```

Comment:
Can you explain more about this situation?

Reply:
not sure what this comment means, but since we now save state metadata even it's null, we should have corresponding parse to deal with null state metadata